### ISSUE TITLE
vips_foreign_load_gif_close() logs error on success

### ISSUE THREAD

**Issue by aaron42net** [jcupitt/libvips#541]
While testing against VIPS 8.4.2 on Debian Jessie with libgif4 v4.1.6-11+deb8u1, I noticed that I was seeing a lot of "gifload_buffer: Unknown error" in the error log that wasn't present with VIPS 8.3.3.

I believe this is a result of vips_foreign_load_gif_close() incorrectly checking the return code of DGifCloseFile(), which returns 1 (GIF_OK) on success and 0 (GIF_ERROR) on error:

```
static void
vips_foreign_load_gif_close( VipsForeignLoadGif *gif )
{
#ifdef HAVE_GIFLIB_5
        if( gif->file ) {
                int error;

                if( DGifCloseFile( gif->file, &error ) )
                        vips_foreign_load_gif_error_vips( gif, error );
                gif->file = NULL;
        }
#else 
        if( gif->file ) {
                if( DGifCloseFile( gif->file ) )
                        vips_foreign_load_gif_error_vips( gif, GifLastError() );
                gif->file = NULL;
        }
#endif

        VIPS_FREEF( fclose, gif->fp );
}
```

I think vips_foreign_load_gif_error_vips() only gets called in both cases on success. Adding a ! in the ifs should fix that, though maybe switching all calls to DGif* to comparing with GIF_ERROR would be more clear.

**Comment by jcupitt** [comment]
Oh good catch. I agree, that's a bug, I'll fix it.

There's supposed to be a thing in the test suite to spot things like this, I wonder why it didn't trigger. I'll investigate.

**Comment by jcupitt** [comment]
Oh, it's to do with previous gif errors not being cleared. Anyway, fixed in master and in 8.4.3. Thank you very much @aaron42net !
